src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -137,6 +140,196 @@ static void verify(ModuleSymbol m)
         }
     }
 
+    [Fact]
+    public void ManuallyAppliedAttribute()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            [RequiresLocation] class C
+            {
+                void M1([RequiresLocation] ref readonly int p) { }
+                void M2([RequiresLocation] in int p) { }
+                void M3([RequiresLocation] ref int p) { }
+                void M4([RequiresLocation] int p) { }
+                [return: RequiresLocation] int M5() => 5;
```

Comment:
>RequiresLocation [](http://example.com/codeflow?start=25&length=16)

Do we allow IsReadOnlyAttribute on return value?

Reply:
No, but that's because `IsReadOnly` can be emitted there by the compiler. `RequiresLocation` is different.